.changeset/eleven-wombats-glow.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/config-loader': minor
+---
+
+**BREAKING** Uses key visibility when the value isn't an array of objects in schema filters. So if the key of an array of string is visible, all elements of this array will also be visible.
```

Comment:
Thank you for looking at this :grin: I actually think this could be a pretty good change, and that we could keep it backwards compatible as well by both looking at the array visibility path and the item visibility path as well.

One ask though, would you be alright with moving this `config-loader` change and opening up a separate PR for it? That way we can discuss and merge these things in isolation